oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/OAuth2UserAuthority.java

```diff
@@ -88,6 +91,39 @@ public OAuth2UserAuthority(String authority, Map<String, Object> attributes, Str
 		this.authority = authority;
 		this.attributes = Collections.unmodifiableMap(new LinkedHashMap<>(attributes));
 		this.userNameAttributeName = userNameAttributeName;
+		this.username = (userNameAttributeName != null && attributes.get(userNameAttributeName) != null)
+				? attributes.get(userNameAttributeName).toString() : null;
+	}
+
+	/**
+	 * Constructs a {@code OAuth2UserAuthority} using the provided parameters. This
+	 * constructor is used by Jackson for deserialization.
+	 * @param authority the authority granted to the user
+	 * @param attributes the attributes about the user
+	 * @param userNameAttributeName the attribute name used to access the user's name from
+	 * the attributes - preserved for backwards compatibility
+	 * @param username the username
+	 */
+	private OAuth2UserAuthority(String authority, Map<String, Object> attributes, String userNameAttributeName,
```

Comment:
since the Builder doesn't use `usernameAttributeName`, we shouldn't include this as a property here, so that this constructor does not deprecated properties.